### PR TITLE
Added an example for encrypting to a group and to a user

### PR DESCRIPTION
**Pull request by coltfred**

### CODE REVIEW

File: examples/encrypting.rs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+/// The group that's encrypted to is created each run, but the user is just the current user.
+#[tokio::main]
+async fn main() -> Result<()> {
+    let (device_context, sdk) =
+        initialize_sdk_from_file(&PathBuf::from("examples/example-ironoxide-user.json")).await?;
+    encrypt_to_group(&sdk).await?;
+    encrypt_to_user(&sdk, device_context.account_id()).await?;
+    Ok(())
+}
+///Create a random group and encrypt the super secret message to it.
```
Comment (giarc3) [inline examples/encrypting.rs:15]: In the most nit-picky way possible, a space after the `///` would make this more consistent.

File: examples/encrypting.rs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        let device_context_file = File::open(&device_path)?;
+        let device_context: DeviceContext = serde_json::from_reader(device_context_file)?;
+        let ironoxide = ironoxide::initialize(&device_context, &Default::default()).await?;
+        Ok((device_context, ironoxide))
+    } else {
+        panic!(format!(
+            "Couldn't open file {} containing DeviceContext",
+            device_path.display()
+        )
+        .to_string(),)
```
Comment (giarc3) [inline examples/encrypting.rs:59]: This `to_string(),` could be removed.

File: examples/example-ironoxide-user.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+  "accountId": "example-ironoxide-user",
+  "segmentId": 1825,
+  "signingPrivateKey": "IW67bY+Yee6oLV1T/AlS82bSP7d9ZqDH3EhmRGJj9js06T9Rx2SjAI3FRUf9O7sflOJ61gDpXVrAO/sNfGZoXw==",
+  "devicePrivateKey": "W7DvCJvGsnXunza+qstf/FTLW42h/VpWN8Be/CkClhY="
+}
```
Comment (giarc3) [inline examples/example-ironoxide-user.json:6]: Where is this device from?

File: examples/encrypting.rs (hunk 2 above)
Comment (coltfred) [reply examples/encrypting.rs:59]: The format! itself can actually be removed, thanks for making me think about that harder.

File: examples/example-ironoxide-user.json (hunk 3 above)
Comment (coltfred) [reply examples/example-ironoxide-user.json:6]: Generated using the production pem file in the tests. We can't comment the json file and keep it json. Do you think it's important to comment on that?

Comment (giarc3) [reply examples/example-ironoxide-user.json:6]: No, I don't think we need to comment it. We've just recently been using a specific project for public devices in tests, and we haven't made ironoxide's public yet, so I was wondering where you decided to get it from.

File: examples/encrypting.rs [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+}
+/// Create a group, asking the SDK to generate a unique ID for it.
+async fn encrypt_to_group(sdk: &IronOxide) -> Result<DocumentId> {
+    // start-snippet{encryptToGroup}
+    let group_id = create_group(sdk).await?;
+    let message = "This is my secret which a whole group should see.";
+    let encrypted_result = sdk
+        .document_encrypt(
+            message.as_bytes(),
+            &DocumentEncryptOpts::with_explicit_grants(None, None, true, vec![(&group_id).into()]),
```
Comment (clintfred) [inline examples/encrypting.rs:23]: An alternative would be to `group_id.clone()` here

Comment (coltfred) [reply examples/encrypting.rs:23]: I would still need the into(), so I'll stick with the borrow.

File: examples/encrypting.rs [hunk 5]
```diff
@@ -0,0 +1,66 @@
+use anyhow::Result;
+use ironoxide::prelude::*;
+use std::{fs::File, path::PathBuf};
+
+/// This is a very basic example of encrypting to a user_id and a group_id.
+/// The group that's encrypted to is created each run, but the user is just the current user.
+#[tokio::main]
+async fn main() -> Result<()> {
+    let (device_context, sdk) =
+        initialize_sdk_from_file(&PathBuf::from("examples/example-ironoxide-user.json")).await?;
```
[the remark this answers is not in the document]
Comment (coltfred) [reply examples/encrypting.rs:10]: true.

**Pull request merged**